irods/test/exception_test.py

```diff
@@ -0,0 +1,50 @@
+#! /usr/bin/env python
+from __future__ import print_function
+from __future__ import absolute_import
+from datetime import datetime
+import errno
+import os
+import re
+import sys
+import unittest
+import irods.test.helpers as helpers
+import irods.exception
+
+
+class TestException(unittest.TestCase):
+
+    def setUp(self):
+        # open the session (per-test)
+        self.sess = helpers.make_session()
+
+    def tearDown(self):
+        # close the session (per-test)
+        self.sess.cleanup()
+
+    def test_400(self):
+        ses = self.sess
+        data = ''
+        try:
+            seed = helpers.my_function_name() + ":" + str(datetime.now())
+            data = helpers.home_collection(ses) + "/" + helpers.unique_name(seed,'data')
+
+            with helpers.create_simple_resc(self,vault_path = '/home') as resc_name:
+                try:
+                    # iRODS will attempt to make a data object where it doesn't have permission
+                    ses.data_objects.create(data,resource = resc_name)
```

Comment:
Should we add an `else` or a `self.fail` on the line after this to ensure that the test does not continue in the event that an `Exception` is somehow not raised? Otherwise, it will result in an error on line 37 rather than a test failure.

Reply:
I suppose it is better that way, yes.  Minimize chance of errors in tests. And, I notice `exc` isn't initialized, which could be bad for the future.